pkg/syncer/spec/mutators/secrets.go

```diff
@@ -48,18 +49,23 @@ func (sm *SecretMutator) Mutate(downstreamObj *unstructured.Unstructured) error
 		return err
 	}
 
-	// We need to transform the default kcp token into an Opaque secret, in order to avoid the pcluster to rewrite it.
-	if secret.Name == "kcp-default-token" {
+	// We need transform the serviceaccount tokens into an Opaque secret, in order to avoid the pcluster to rewrite them,
+	// and we remove the annotations that point to the kcp serviceaccount name/uid.
+	if _, ok := secret.Annotations[core.ServiceAccountNameKey]; ok {
 		secret.Type = corev1.SecretTypeOpaque
+		annotations := secret.GetAnnotations()
+		delete(annotations, core.ServiceAccountNameKey)
+		delete(annotations, core.ServiceAccountUIDKey)
+		secret.SetAnnotations(annotations)
```

Comment:
if you use a secret (i.e. non-unstructured), you can mutate the Annotation directly. No need to get and set.